### ISSUE TITLE
Nested (!) scoped threadpools silently not executing work?

### ISSUE THREAD

**Issue by fritshoogland-yugabyte** [rayon-rs/rayon#972]
I must say I am not sure if rayon is involved here, but so far it looks like it.

I created a program where there are multiple units of work executed via a rayon scoped pool:
```
    let main_pool = rayon::ThreadPoolBuilder::new().num_threads(parallel).build().unwrap();
    main_pool.scope(|mps| {
       mps.spawn(move |_| {
           metric::perform_metrics_snapshot( .... );
        }
        ....
    }
```
Inside these functions, another scoped thread pool is created in which each perform work via reqwest:blocking::get() for a number of ip addresses is performed.

When I set the parallel variable to 1, it seems I do not get data via reqwest for a specific single request(!). If I set the parallel parameter from 1 to 3, I do get response from all IP address:port combinations.

I am not even sure if nested scoped thread pools are supported and supposed to be working.
There are no error messages, reqwest simple does not return anything for a single IP address:port.

**Comment by fritshoogland-yugabyte** [comment]
The error was on my side, I am sorry to waste your time and some very fine CPU cycles!

**Comment by cuviper** [comment]
No worries! I'm curious what it was... But just to confirm, yes we do support nested use of threadpools.

**Comment by fritshoogland-yugabyte** [comment]
The application I am building creates a scoped thread pool that spawns a thread for every distinct endpoint for a group of servers. That is done in the main scoped thread group.
Inside the function that performs the fetching, I iterate over a group of hosts and ports, which is done in another scoped thread group, because I do suspect that most of the time is waiting for network IO, so I can do some of that in parallel too, to speed up the fetching.

Having done all that, I started missing a specific endpoint, quite randomly, which seemed to get solved when I increased the amount of parallelism, hence the question.

But when I investigated it, it turned out that the fetch of the endpoint was still done, it was the parsing that failed to produce the results in the struct. And that was caused by fields that are not always there, and thus needed to be wrapped in an <option>.
